source/develop/release-management/features/virt/nvdimm-device.md

```diff
@@ -0,0 +1,67 @@
+---
+title: NVDIMM devices
+category: feature
+authors: mzamazal
+feature_name: NVDIMM devices
+feature_modules: engine,vdsm
+feature_status: In development
+---
+
+# NVDIMM devices
+
+## Summary
+
+This feature allows adding NVDIMM (non-volatile memory) devices to VMs.
+
+## Owner
+
+*   Name: Milan Zamazal
+*   Email: mzamazal@redhat.com
+
+## Description
+
+NVDIMM devices provide persistent RAM.  QEMU and libvirt allow to add an emulated NVDIMM device to a VM, backed by a real NVDIMM device on the host to ensure data persistence.
+
+This feature allows assigning host NVDIMM devices to VMs.  A new NVDIMM category of host devices is added (although technically NVDIMM devices are treated as memory rather than host devices by libvirt).
+
+The NVDIMM device in the guest is emulated so its mode, partitioning, etc. can be reconfigured in the guest independently of the host device settings.
+
+Information about the emulation details can be found in
+[QEMU NVDIMM documentation](https://github.com/qemu/qemu/blob/master/docs/nvdimm.txt).
+
+## Prerequisites
+
+NVDIMM hardware is needed to test this feature.  To a limited extent, QEMU emulated devices in VMs used as hosts can be used for testing during implementation.
+
+## Limitations
+
+Memory snapshots are disabled when an NVDIMM device is present in a VM.  There is no way to make a snapshot of NVDIMM content and a memory snapshot cannot work correctly without having the corresponding NVDIMM data.  If a memory snapshot is attempted, a non-memory snapshot is created instead.
+
+In oVirt, each NVDIMM device passed to a VM has an automatically assigned label area of a fixed size of 128 KB.  Having a label is required on POWER and 128 KB is the minimum label size allowed by QEMU.
+
+The guest device size is not user configurable.  If the whole host NVDIMM device size should not be used, the device should be partitioned accordingly on the host and the corresponding partition should be used.
+
+The guest device size may be a bit lower to be in compliance with libvirt and QEMU alignment and size adjustments.  Precise sizing is also needed in order to make memory hot plug work.
+
+libvirt and QEMU perform their own size and label placement adjustments.  If those internal arrangements ever get changed, data loss may be experienced.
+
+NVDIMM hot plug is not supported.
```

Comment:
by the platform, right? worth mentioning that it's not a "management gap"

Reply:
Done.